File: source/sat/src/main/java/ch/unisg/sat/controller/http/dto/response/ad/ArchitectureRationaleResponseDto.java
package ch.unisg.sat.controller.http.dto.response.ad;

import ch.unisg.sat.core.domain.entities.classes.ad.ArchitectureRationale;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.UUID;

public class ArchitectureRationaleResponseDto {

    private static final String LOCATION = "http://localhost:4000/api/v1/";

    public static String uri(UUID id) {
        return LOCATION + id.toString();
    }

    public static List<HashMap<String, Object>> toJson(List<ArchitectureRationale> architectureRationales) {

        List<HashMap<String, Object>> architectureRationalesResponseDto = new ArrayList<>();

        architectureRationales.forEach(architectureRationale -> {
            HashMap<String, Object> architectureRationaleResponseDto = new HashMap<>();

            architectureRationaleResponseDto.put("id", architectureRationale.getId());
            architectureRationaleResponseDto.put("title", architectureRationale.getTitle());
            architectureRationaleResponseDto.put("context", architectureRationale.getContext());
            architectureRationaleResponseDto.put("decision", architectureRationale.getDecision());
            architectureRationaleResponseDto.put("status", architectureRationale.getStatus());
            architectureRationaleResponseDto.put("consequences", architectureRationale.getConsequences());

            architectureRationalesResponseDto.add(architectureRationaleResponseDto);
        });

        return architectureRationalesResponseDto;

    }

    public static HashMap<String, Object> create(ArchitectureRationale architectureRationale) {

        HashMap<String, Object> result = new HashMap<>();

        result.put("id", architectureRationale.getId().toString());
        result.put("title", architectureRationale.getTitle());

        return result;
    }
}

File: source/frontend/src/lib/dto/response/entity/ArchitectureRationaleResponseDto.ts
import {Rationale} from "$lib/domain/entity/ad/Rationale.svelte";

export class RationaleResponseDto {
    id : string = "";
    title : string = "";
    context : string = "";
    decision : string = "";
    status : string = "";
    consequences : string = "";
    justifies : string = "";
}

export function fromRationalesResponseDto(
    architecturalRationalesResponseDto : RationaleResponseDto[]
) : Rationale[] {
    return architecturalRationalesResponseDto.map(fromRationaleResponseDto);
}

export function fromRationaleResponseDto(
    rationaleResponseDto : RationaleResponseDto
) : Rationale {
    const rationale : Rationale = Rationale.create();

    rationale.id = rationaleResponseDto.id;
    rationale.title = rationaleResponseDto.title;
    rationale.context = rationaleResponseDto.context;
    rationale.decision = rationaleResponseDto.decision;
    rationale.status = rationaleResponseDto.status;
    rationale.consequences = rationaleResponseDto.consequences;
    rationale.justifies = rationaleResponseDto.justifies;

    return rationale;
}